src/gil.rs

```diff
@@ -64,13 +64,8 @@ pub fn prepare_freethreaded_python() {
         if ffi::Py_IsInitialized() != 0 {
             // If Python is already initialized, we expect Python threading to also be initialized,
             // as we can't make the existing Python main thread acquire the GIL.
-            #[cfg(not(Py_3_7))]
-            assert_ne!(ffi::PyEval_ThreadsInitialized(), 0);
+            debug_assert_ne!(ffi::PyEval_ThreadsInitialized(), 0);
```

Comment:
I think `assert_ne` is better than `debug_assert_ne` here:
- it's not a hot code path (only called once), so the performance overhead doesn't really matter in `--release`
- code will crash in weird ways if threads are not initialised.

Reply:
👍🏼 , I changed to use `assert_ne`